evm-adapters/src/evm_opts.rs

```diff
@@ -153,7 +154,7 @@ pub struct Env {
     pub gas_limit: u64,
 
     /// the chainid opcode value
-    pub chain_id: u64,
+    pub chain_id: Option<u64>,
```

Comment:
makes sense to make this optional,

so that default is we detect it via RPC?

Reply:
It defaults to 1 in all cases except if we are forking a remote RPC, in that case it defaults to whatever the RPC reports the chain ID to be